processor/probabilisticsamplerprocessor/sampler_mode.go

```diff
@@ -293,12 +479,18 @@ func commonShouldSampleLogic[T any](
 	logger *zap.Logger,
 ) bool {
 	rnd, carrier, err := randFunc(item)
+
 	if err == nil {
 		err = consistencyCheck(rnd, carrier)
 	}
 	var threshold sampling.Threshold
 	if err != nil {
-		logger.Debug(description, zap.Error(err))
+		var se samplerError
+		if errors.As(err, &se) {
+			logger.Info(description, zap.Error(err))
```

Comment:
this is part of the hot path, I'd rather have a metric recording the number of errors that happened, with samplerError being a dimension of the metric (error=sampler, error=other).

Reply:
Let me suggest something different: let them all be (debug) logs for this PR, but then convert the metrics for this component as a whole from OC to OTel, using mdatagen. On that PR, you could add the new metric for the error conditions.

Reply:
I like the idea of making the logs into Debug() and Info()-level. I also do not expect any of the Info()-level log statements to be common, for they require a misbehaving SDK or some other form of data corruption, as opposed to misconfigured sampling. Because these messages signal corruption of some kind, I think they should be logs. My expectation is that logs are sampled at runtime (or they should be) to avoid impacting the hot path.

With that said, I feel that a new metric is not helpful -- it just means the customer has to monitor something new when we have metric signals already. There is a single metric output by this code specifically, which will have a "policy=missing_randomness" when these errors arise.

We also [have (or at least desire) standard pipeline metrics](https://github.com/open-telemetry/oteps/pull/259), which ought to provide a standard way to count how many spans succeed or fail. If the sampler is configured with FailClosed=true and these missing_randomness conditions are happening, the result will be loss of spans. I do not want the user to have to discover a new metric for this, because there ought to be a standard metric for rejected items. All processors should be able to count the number of items that are rejected for malformed data.